tests/conftest.py

```diff
@@ -7092,6 +7092,112 @@ def teardown():
     return factory
 
 
+@pytest.fixture
+def multi_cnv_workload(
+    pv_encryption_kms_setup_factory, storageclass_factory, cnv_workload
+):
+    """
+    Create a cnv factory. Calling this fixture Creates multiple VMs
+    with specified configurations using the cnv_workload fixture.
+    """
+
+    def factory(namespace=None):
+        """
+        Args:
+            namespace (str, optional): The namespace to create the vm on.
+
+        Returns:
+            lists: objects of cnv workload class with default comp and aggressive compression
+
+        """
+        vm_list_agg_compr = []
+        vm_list_default_compr = []
+
+        namespace = (
+            namespace if namespace else create_unique_resource_name("vm", "namespace")
+        )
+
+        # Setup csi-kms-connection-details configmap
+        log.info("Setting up csi-kms-connection-details configmap")
+        kms = pv_encryption_kms_setup_factory(kv_version="v2")
+        log.info("csi-kms-connection-details setup successful")
+
+        # Create an encryption enabled storageclass for RBD
+        sc_obj_def_compr = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=kms.kmsid,
+            new_rbd_pool=True,
+        )
+
+        sc_obj_aggressive = storageclass_factory(
+            interface=constants.CEPHBLOCKPOOL,
+            encrypted=True,
+            encryption_kms_id=kms.kmsid,
+            compression="aggressive",
+            new_rbd_pool=True,
+        )
+        vm_configs = [
```

Comment:
Please make sure you cover all RWX and at least one RWO configs from [here](https://docs.google.com/spreadsheets/d/1KX6WueORPjaOlInTLMAl1Baga64K2k_ELFGGOb7qKh0/edit?gid=269509650#gid=269509650&range=I99:R108)